lib/libYulAST/IntrinsicPatterns.cpp

```diff
@@ -198,4 +210,30 @@ IntrinsicPatternMatcher::parseYulStructType(std::string_view name) {
   return res;
 }
 
+ArrayTypeResult IntrinsicPatternMatcher::parseArrayType(std::string_view name) {
+  ArrayTypeResult res{};
+  std::string nameStr(name);
+  std::regex regex(ARRAY_TYPE_REGEX_LIT);
+  std::smatch match;
+  int match_success = std::regex_match(nameStr, match, regex);
```

Comment:
`std::regex_match` returns a bool:
```suggestion
  bool match_success = std::regex_match(nameStr, match, regex);
```

Reply:
done,